js/background.js

```diff
@@ -1,80 +1,76 @@
 /* This script runs in the background from when the app is loaded */
 
-var Domain = "https://channeli.in";
-var Host = "channeli.in";
+var NET_CHECK_DOMAIN = "http://info.cern.ch/";
+var DOMAIN = "https://channeli.in";
+var HOST = "channeli.in";
 
-var getDomainName = function (href) {
-  var l = document.createElement("a");
-  l.href = href;
-  return l.hostname;
-}
+/** 0 implies 'not logged in' and 1 implies 'logged in' */
+var userStatus = -1;
+/** 0 implies 'not connected' and 1 implies 'connected' */
+var networkStatus = 0;
 
-var checkSession = function() {
-  var url = Domain + "/check-session/";
-  $.post(url, function(res){
-    if(res.msg == "YES") {
-      chrome.browserAction.setIcon({path: "../images/icon_active.png"});	
-    }
-    else if(res.msg == "NO") {
-      chrome.browserAction.setIcon({path: "../images/icon_inactive.png"});
+/**
+ * Check session if the changed tab was one of Channel-i
+ * @param tabId - the ID of the tab that is changing or has changed
+ * @param changeInfo - whether the tab is changing or has changed
+ * @param tab - the tab that is changing or has changed
+ */
+function updateListener(tabId, changeInfo, tab) {
+    var url = tab.url;
+    if (url !== undefined && changeInfo.status === "complete") {
+        var host = getHostName(url);
+        if (host == HOST) {
+            checkSession();
+        }
     }
-  });
 }
 
-// This is to handle the status updation when the app is "loaded" or "refreshed". 
-checkSession(); 
+/**
+ * Get the host name from a given URL
+ * @param href - the URL taken from the address bar
+ * @returns {string} hostname - the host name extracted from the URL
+ */
+var getHostName = function (href) {
```

Comment:
use consistent functions. Somewhere its a function expression and somewhere its function declaration. 

function expression:
```js
var x = function(){}
```

function declaration
```js
function x(){}
```